lib/engines/hcktest/hcktest.rb

```diff
@@ -151,21 +151,28 @@ def configure_clients
       end
     end
 
+    def reconfigure_clients
+      configure_clients
+      @clients.each_value(&:synchronize)
+    end
+
     def configure_setup_and_synchronize
+      return reconfigure_clients unless @studio.tools.nil?
+
       @studio.configure(@platform['clients'])
       configure_clients
       @clients.each_value(&:synchronize)
```

Comment:
Call `reconfigure_clients` here (and `reconfigure_clients` should be renamed as it is awkward to call like that when it is used for the first run).